port/raspberrypi/rp2xxx/src/hal/spi.zig

```diff
@@ -162,6 +161,17 @@ pub const SPI = enum(u1) {
         spi_regs.SSPCPSR.modify(.{ .CPSDVSR = 0 });
     }
 
+    pub fn set_slave(spi: SPI, slave: bool) void {
+        const regs = spi.get_regs();
+        // Disable SPI
+        regs.SSPCR1.modify(.{ .SSE = 0 });
```

Comment:
I wonder whether we really need this, since we are writing to the same register, we can do this atomically.

Reply:
I imagine you could probably set MS and enable at the same time. But disabling and enabling doesn't make sense as something you could do atomically.